### DISCUSSION TITLE
Unclear seed entry UI

### DISCUSSION THREAD

**Discussion by johnheenan** [btcpayserver/btcpayserver#6268] [General]
The seed entry UI below is unclear and it is easy to miss that a switch must be toggled to enable hot wallet use. This has already resulted in an issue https://github.com/btcpayserver/btcpayserver/issues/6248 that has given rise to another issue https://github.com/btcpayserver/btcpayserver/issues/6261

![image](https://github.com/user-attachments/assets/b909e4f9-8a00-44db-b2fc-f0b2a104b998)

Extract from https://chat.btcpayserver.org/btcpayserver/pl/dusonoy99tyexcmjcmftgq5o5o

The screen shot above shows poor UI that can be improved. The expectation is that entering seed phrases automatically results in a 'hot wallet'. A clear warning at the top stating 'READ ONLY XPUB EXTRACTED UNLESS YOU TOGGLE IS HOT WALLET' would help.

There is no clear distinction anywhere between 'read only' wallets and 'hot' wallets. There is a distinction in source code and I don't know what the distinction is.

Instead of a single window to 'Enter wallet seed' I think an improvement would be to offer two choices: 'Extract read only xpub from wallet seed' and 'Use wallet seed for hot wallet', if that is what the difference is between toggling 'is hot wallet' on or off means. 

Extract from https://chat.btcpayserver.org/btcpayserver/pl/aok9fipfpbnf3pmrjen8yjdfzc

## My understanding of wallet types.

**Hot Wallet:** Wallet with private keys exposed to a third party for automatic transaction signing and transmitting. Very insecure and never advised except perhaps for small amounts or temporary risk.

**Cold Wallet:** Wallet with private keys protected from third parties with provided mechanism to sign transactions generated by a third party and send signed transaction back to third party for transmitting.

**Read only wallet:** xpub or public key wallet with no provided mechanism for third parties to request and receive a signed transaction back  for transmission.

An xpub or public key wallet can still generate transactions and request they be signed and transmitted elsewhere. 

The distinctions are not clear cut because they are implementation dependent. For example a cold wallet might be usable in a UI but not in an API. More proposals with source code significance. Not sure of their validity.

**Not read only wallet:** Hot wallet or cold wallet. Invalid wallet for direct API use for signing if cold wallet. Yes if directed to a UI.

**Not hot wallet:** Cold wallet or read-only wallet. Invalid wallet for direct API use for signing. Yes if directed to a U

**Comment by pavlenex** [comment]
How is it unclear that you have to enable a switch with hot wallet? What's unclear there, can you be more specific? This is how we refer to wallets throughout our docs and UI. cc @dstrukt

**Comment by johnheenan** [comment]
The only place in the UI I ever remember seeing 'hot wallet' is above. The screen before snapshot above calls a 'cold wallet' a 'hardware wallet' and this is the terminology used in documentation. 

'hotwallet' and 'readonlywallet' appears in BTCPay source code.

Technically the term 'wallet' is meaningless in cryptocurrency. Cryptocurrency only recognises private keys, public keys, signing with private keys, verifying with public keys, seed phrases, pass phrases, derivation paths, xpubs (and equivalent), xprivs (and equivalent). No mention of the word wallet anywhere in technical literature.

There is an expectation that entering a seed phrase results in a 'hot wallet'. In the snapshot image it literally says 'is hot wallet'. So I expect most people, if they even bother to notice, will think they now have got what they expect, which is a 'hot wallet'. A few people might notice there is a disabled button and might make the connection they are expected to toggle the button to make the wallet a 'hot wallet'.

Also the text near the toggle is very confusing and does not even appear accurate.

Surely a better way is to state clearly something like the following, if accurate (and at the top):

"Your seed phrase will be used to extract and store an xpub for reading. To extract and store an xpriv for a 'hot wallet' you must also toggle button below to enabled.  Your seed phrase will not be recorded."

**Comment by pavlenex** [comment]
I still don't understand what you want us to do with this information?
What you're suggesting a non dev user doesn't understand.
Is this already defined somehow, perhaps bitcoin.design? Otherwise it's something we've been using for years and accross our docs, so if you'd want this changed you'd have to not only update UI but also docs, I personally don't see the benefit of this. This is technical vs user-facing terminology.

I suggest this isssue to be moved to discussions, as it's not a bug report.

**Comment by johnheenan** [comment]
No, this it is not a bug report but it is a valid issue and is illustrative.

If you don't agree the UI is confusing and that this particular issue does not merit a quick source code fix here then fine. Close the issue. I am not going to request the issue is reopened.

I do agree there should be a general discussion on making the UI less confusing. This issue is specific, it does not belong in a discussion.

**Comment by johnheenan** [comment]
Sugested fix for issue repeated from above, with slight alteration:

If accurate, add text somewhere near top: "Your seed phrase will be used to extract and store an xpub for reading. To extract and store an xpriv for a 'hot wallet' you must also toggle switch below to on. Your seed phrase will not be recorded."

Also, at the very least, remove the words "as metadata" from the toggle switch message. It adds nothing except confusion. 

Even the term 'checked' for the toggle switch is confusing. It is not a check box. Why not use 'toggled on' instead?

**Comment by dstrukt** [comment]
Hey @johnheenan thanks for this feedback, there's always more we can do to improve these wallet configuration flows, and it has been on the list of priorities for us. 

I'll work your feedback into some of the updated mocks and follow up with you to see if it addresses your concerns!

**Comment by pavlenex** [comment]
@johnheenan, we use issues primarily for bug reports and actionable tasks with clear paths to resolution. I appreciate your feedback and respect if you have a different approach to issue management. However, I kindly ask that you pay closer attention to your communication, as your tone and bypassing the issue templates make it more difficult to understand and address your concerns efficiently. This creates additional steps and slows down the process.

**Comment by johnheenan** [comment]
> However, I kindly ask that you pay closer attention to your communication, as your tone and bypassing the issue templates 

I presented a complex issue in a valid manner, with context  and suggested a solution right from the first post. Full stop.

You are now publically attacking my abilty to present an issue, my communication and my tone, apparently solely on the basis I did not fill in your template. Is this correct? Because I am clueless about what I am doing wrong, in your view.

**Comment by pavlenex** [comment]
I don't think you're even reading what I wrote, if you want to discuss this feel free to ping me on our Mattermost. I am not going to allow you to play victimizaiton card here just because I've asked you to be respectful and follow process we have for filing the issue. Since this discussion is no longer related to anything,or productive if you wish to continue or more or need clarity dm me.
